iep-module-userservice/src/main/java/com/netflix/iep/userservice/HttpUserService.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2014-2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.iep.userservice;
+
+import com.github.benmanes.caffeine.cache.Caffeine;
+import com.github.benmanes.caffeine.cache.LoadingCache;
+import com.netflix.iep.service.AbstractService;
+import com.netflix.spectator.ipc.http.HttpResponse;
+import com.typesafe.config.Config;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.net.URI;
+import java.util.Collections;
+import java.util.List;
+import java.util.Locale;
+import java.util.Set;
+
+/**
+ * Calls one or more remote services via HTTP to check if an email is valid. The only
+ * supported call is {@link #isValidEmail(String)}, the others just return an empty set.
+ * Emails checks will get cached to reduce load and improve performance for repeated checks.
+ */
+@Singleton
+public class HttpUserService extends AbstractService implements UserService {
+
+  private final Context context;
+  private final boolean enabled;
+  private final List<? extends Config> uris;
+
+  private final LoadingCache<String, Boolean> cache;
+
+  @Inject
+  HttpUserService(Context context) {
+    this(context, "http");
+  }
+
+  HttpUserService(Context context, String name) {
+    this.context = context;
+
+    Config config = context.config().getConfig(name);
+    this.enabled = config.getBoolean("enabled");
+    this.uris = config.getConfigList("uris");
+
+    this.cache = Caffeine.newBuilder()
```

Comment:
It occurred to me last night that we should add a cache to the implementation - nice catch!